data_source.go

```diff
@@ -345,19 +367,18 @@ func (ds *AnySource) PrepareRun() error {
 		dsp.stream.voltsPerArb = vpa[channelNum]
 		ds.processors[channelNum] = dsp
 
-		// TODO: don't just set these to arbitrary values
+		ts := tsptrs[channelNum]
+		if ts == nil {
+			ts = &defaultTS
+		}
+		dsp.TriggerState = *ts
+
+		// TODO: don't just set decimation and samples/presamples to arbitrary values
 		dsp.Decimate = false
-		dsp.DecimateLevel = 3
+		dsp.DecimateLevel = 2
 		dsp.DecimateAvgMode = true
-		dsp.LevelTrigger = false
-		dsp.LevelLevel = 4000
-		dsp.EdgeTrigger = true
-		dsp.EdgeLevel = 100
-		dsp.EdgeRising = true
-		dsp.NPresamples = 200
+		dsp.NPresamples = 250
 		dsp.NSamples = 1000
```

Comment:
Can we save NSamples and NPresamples as well? Maybe in a different PR, but a keything to minimize steps for doing the same thing every time.

Reply:
Actually, I think these ARE already saved. Let me figure this out...